docs/proposals/service-binding-specification/implementing-specification.md

```diff
@@ -0,0 +1,129 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for alpha, beta, GA maturity
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding to implement the
+design. For instance:
+
+> 1. This prevents us from doing cross-namespace binding. Should we do this?
+
+## Summary
+
+This document proposes a mechanism for the Service Binding Operator to implement the
+[Service Binding Specification for Kubernetes](https://github.com/k8s-service-bindings/spec).
+
+## Motivation
+
+The [Service Binding Specification for Kubernetes](https://github.com/k8s-service-bindings/spec)
+aims to create a mechanism that is widely applicable. The benefit of a Kubernetes-wide service binding
+specification is that all of the actors in an ecosystem, including application developers and service
+providers, can work towards a clearly defined abstraction at the edge of their expertise and depend on
+other parties to complete the chain.
+
+On the other hand, the Service Binding Operator has a well-defined mechanism for binding applications
+and services together. This mechanism is different from how the Service Binding Specification for
+Kubernetes lays out binding between entities. Generally, most features defined in the specification
+are a subset of features defined in the Service Binding Operator.
+
+There are major differences between the specification and the current mechanism the Service Binding
+Operator employs such as how the application is defined in the CRD, how a binding secret is
+projected in an application container and etc. The [Service Binding feature comparison document](./service-binding-feature-comparison.pdf) lists the differences between the two.
+
+One of the major gaps between the CRDs defined by the two approaches is that the Service Binding
+Operator supports defining multiple services within a CR. Whereas the specification allows defining
+one service per CR. Here are two main reasons why the specification is in favor of defining one
+service per CR:
+
+- It is easier for tools created around the service binding to add and remove services by simply
+  adding or removing independent CRs instead of having to deal with editing an array of services
+  within a CR. In addition, editing the service array in a CR might cause CR to be corrupted unless
+  concurrent updates are well synchronized across a tool or tools. However, by defining a single
+  service per CR, adding or removing services simply becomes `kubectl apply ...` and
+  `kubectl delete ...`.
+
+- Defining one service per CR allows a cleaner association of attributes with the service. For example,\
+  when we identify a set of extra mappings or environment variables in a `ServiceBinding`, which
```

Comment:
Thank you for adding details.

> For example,\
  when we identify a set of extra mappings or environment variables in a `ServiceBinding`, which
  service does that get associated with if we have multiple services? Or if we want to change the
  service name or type that gets mounted, we again run into the issue that they all require a way
  to identify the service you want to apply those attributes against.

The existing implementation doesn't have this issue.

(1) you can unique identify keys coming from the a specific service by using service specific prefixes.
(2) you can use custom env vars where you can refer to the group.version.kind.name.service_specific_key

Reply:
I think there are 2 important points where the spec differs over a multi-service implementation:

(1) the `ServiceBinding` extra mappings become more readable because you keys are referenced simply by using `key` instead of `group.version.kind.name.service_specific_key`
(2) the `ServiceBinding` extra mappings are volume mounted with the same single Secret of the CR.  In the case of multiple services with `ServiceBindingRequest`, where do the extra mappings get volume mounted under?  (in the scenario where volume mount is picked instead of env vars)

Reply:
>  the ServiceBinding extra mappings become more readable because you keys are referenced simply by using key instead of group.version.kind.name.service_specific_key

You could alias it as "x" or even use the service id "x" to do a x.username , y.password . We currently support 
* Aliases in the golang template 
* service id https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/apis/operators/v1alpha1/servicebinding_types.go#L59
* the full g.v.k.name.something 

> where do the extra mappings get volume mounted under

There are two levels of prefixes for both env var and vol paths: 
* Service specific, or 
* Global. In this case, this would be a service specific volume mount path prefix. This isn't implemented yet on a service level, but should be do-able given the direction things are going in.